### PR TITLE
Ensure DeprecationErrorHandler::collectDeprecations() is triggered

### PR DESCRIPTION
**Pull request by alexpott**
Ensure DeprecationErrorHandler::collectDeprecations() is triggered if SYMFONY_DEPRECATIONS_SERIALIZE is set

| Q             | A
| ------------- | ---
| Branch?       | 3.3 
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes 
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Drupal doesn't use src/Symfony/Bridge/PhpUnit/bin/simple-phpunit but would like to use the deprecation collection features of src/Symfony/Bridge/PhpUnit. The checks in src/Symfony/Bridge/PhpUnit/bootstrap.php mean that this is difficult because they rely on simple-phpunit - but I'm not sure that that is necessary.

The problem is even in isolated tests we have PHPUNIT_COMPOSER_INSTALL set because we use a custom phpunit config file.

### CODE REVIEW

**Comment by alexpott** [comment]
We tried this before in https://github.com/symfony/symfony/pull/24685 but got it wrong.

**Comment by alexpott** [comment]
I've run src/Symfony/Component/Process/Tests/ProcessTest.php locally and this version does not seem to cause the same problems

File: src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php [hunk 1]
```diff
@@ -262,6 +262,7 @@ public function endTest($test, $time)
         if ($this->runsInSeparateProcess) {
             $deprecations = file_get_contents($this->runsInSeparateProcess);
             unlink($this->runsInSeparateProcess);
+            putenv('SYMFONY_DEPRECATIONS_SERIALIZE=');
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php]: the `=` should be removed, to really remove the env var (the `=` makes it the empty string, where we want to delete it, isn't it?)

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexpott.

**Pull request closed without merging**